tests/lib/api.js

```diff
@@ -37,4 +42,49 @@ describe("api", () => {
     it("should have SourceCode exposed", () => {
         assert.isFunction(api.SourceCode);
     });
+
+    describe("loadESLint", () => {
+
+        afterEach(() => {
+            delete process.env.ESLINT_USE_FLAT_CONFIG;
+        });
+
+        it("should be a function", () => {
+            assert.isFunction(api.loadESLint);
+        });
+
+        it("should return a Promise", () => {
+            assert.instanceOf(api.loadESLint(), Promise);
+        });
+
+        it("should return FlatESLint when useFlatConfig is true", async () => {
+            assert.strictEqual(await api.loadESLint({ useFlatConfig: true }), FlatESLint);
+        });
+
+        it("should return ESLint when useFlatConfig is false", async () => {
+            assert.strictEqual(await api.loadESLint({ useFlatConfig: false }), api.ESLint);
+        });
+
+        it("should return FlatESLint when useFlatConfig is not provided because we have eslint.config.js", async () => {
+            assert.strictEqual(await api.loadESLint(), FlatESLint);
+        });
+
+        it("should return ESLint when useFlatConfig is not provided because we have .eslintrc.cjs", async () => {
+            assert.strictEqual(await api.loadESLint({
+                cwd: "./tests/fixtures/config-file/cjs"
+            }), FlatESLint);
+        });
```

Comment:
According to its title, this test expects `ESLint`, but checks if `FlatESLint` was returned? Also, the presence of `.eslintrc.cjs` doesn't affect the logic.

Perhaps we want to test something like this:

```suggestion
        it("should return ESLint when useFlatConfig is not provided and there is no eslint.config.js", async () => {
            assert.strictEqual(await api.loadESLint({
                cwd: os.tmpdir()
            }), api.ESLint);
        });
```